managers/bosh-manager/DirectorService.js

```diff
@@ -0,0 +1,1112 @@
+'use strict';
+
+const assert = require('assert');
+const Promise = require('bluebird');
+const _ = require('lodash');
+const yaml = require('js-yaml');
+const config = require('../../common/config');
+const logger = require('../../common/logger');
+const errors = require('../../common/errors');
+const jwt = require('../../broker/lib/jwt');
+const utils = require('../../common/utils');
+const catalog = require('../../common/models').catalog;
+const NotFound = errors.NotFound;
+const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
+const ScheduleManager = require('../../jobs');
+const CONST = require('../../common/constants');
+const ordinals = ['First', 'Second', 'Third', 'Fourth', 'Fifth', 'Sixth', 'Seventh', 'Eighth', 'Ninth', 'Tenth'];
+const bosh = require('../../data-access-layer/bosh');
+const Agent = require('../../data-access-layer/service-agent');
+const NetworkSegmentIndex = bosh.NetworkSegmentIndex;
+const ServiceBindingAlreadyExists = errors.ServiceBindingAlreadyExists;
+const backupStore = require('../../data-access-layer/iaas').backupStore;
+const boshOperationQueue = bosh.BoshOperationQueue;
+const ServiceInstanceAlreadyExists = errors.ServiceInstanceAlreadyExists;
+const ServiceInstanceNotOperational = errors.ServiceInstanceNotOperational;
+const FeatureNotSupportedByAnyAgent = errors.FeatureNotSupportedByAnyAgent;
+const ServiceBindingNotFound = errors.ServiceBindingNotFound;
+const DeploymentDelayed = errors.DeploymentDelayed;
+const BaseDirectorService = require('../BaseDirectorService');
+const cf = require('../../data-access-layer/cf');
+const cloudController = cf.cloudController;
+const serviceFabrikClient = cf.serviceFabrikClient;
+const Header = bosh.manifest.Header;
+const Addons = bosh.manifest.Addons;
+const EvaluationContext = bosh.EvaluationContext;
+const BadRequest = errors.BadRequest;
+const BasePlatformManager = require('../../broker/lib/fabrik/BasePlatformManager');
+
+
+class DirectorService extends BaseDirectorService {
+  constructor(guid, plan) {
+    super(plan);
+    this.guid = guid;
+    this.plan = plan;
+    this.cloudController = cloudController;
+    this.serviceFabrikClient = serviceFabrikClient;
+    this.director = bosh.director;
+    this.networkSegmentIndex = undefined;
+    this.platformManager = undefined;
+    this.backupStore = backupStore;
+    this.agent = new Agent(this.settings.agent);
+    this.prefix = CONST.SERVICE_FABRIK_PREFIX;
+  }
+
+  assignPlatformManager(platformManager) {
+    this.platformManager = platformManager;
+  }
+
+  get platformContext() {
+    return Promise.try(() => this.networkSegmentIndex ? this.deploymentName : this.director.getDeploymentNameForInstanceId(this.guid))
+      .then(deploymentName => this.director.getDeploymentProperty(deploymentName, CONST.PLATFORM_CONTEXT_KEY))
+      .then(context => JSON.parse(context))
+      .catch(NotFound, () => {
```

Comment:
By now all deployments should have this property. Is this code still required?

Reply:
Kept it as original. :)